packages/babel-plugin-minify-simplify/__tests__/simplify-test.js

```diff
@@ -2136,4 +2136,57 @@ describe("simplify-plugin", () => {
     const expected = source;
     expect(transform(source)).toBe(expected);
   });
+
+  it("should transform assignments to the same identifier", () => {
+
+    const actual = [
+      'x = x + 1;',
+      'x = x - 1;',
+      'x = x * 1;',
+      'x = x % 1;',
+      'x = x << 1;',
+      'x = x >> 1;',
+      'x = x >>> 1;',
+      'x = x & 1;',
+      'x = x | 1;',
+      'x = x ^ 1;',
+      'x = x / 1;',
+      'x = x ** 1;',
+      'foo = foo + bar;',
+      'foo = foo * function(){};',
+      'foo += 123;',
+      'foo = 1 + foo;',
+      'x = x = x + 1;',
+      'foo = foo + bar + baz;',
+      'window.foo = window.foo + 1;',
+      'window.foo.bar = window.foo.baz + 1;',
+    ];
+
+    const expected = [
+      'x++;',
+      'x--;',
+      'x *= 1;',
+      'x %= 1;',
+      'x <<= 1;',
+      'x >>= 1;',
+      'x >>>= 1;',
+      'x &= 1;',
+      'x |= 1;',
+      'x ^= 1;',
+      'x /= 1;',
+      'x **= 1;',
+      'foo += bar;',
+      'foo *= function () {};',
+      'foo += 123;',
+      'foo = 1 + foo;',
+      'x = x++;',
+      'foo = foo + bar + baz;',
+      'window.foo = window.foo + 1;',
+      'window.foo.bar = window.foo.baz + 1;',
+    ];
+
+    actual.forEach((test, index) => {
+      expect(transform(test)).toBe(expected[index]);
```

Comment:
```js
expect(actual.map((s) => transform(s))).toEqual(expected);
```

here you compare two arrays and even if one fails in the list, the others still run. But when you do `forEach` when one of them fails, all else that follows won't be run at all.